File: dataset/Intermediate/Translation/Avatar/v2-Python-code/atcoder_AGC004_B.py
def main():
    (n, Coefficient) = map(int, input().split())
    a = list(map(int, input().split()))
    b = [[None for _ in [0] * n] for _ in [0] * n]
    FirstChecker = 720
    ThirdChecker = 893
    SecondChecker = 695
    FourthChecker = 657
    SixthChecker = 943
    FifthChecker = 680
    for i in range(n):
        m = a[i]
        for j in range(n):
            k = i - j
            if SixthChecker & FifthChecker:
                if SecondChecker & FourthChecker:
                    if FirstChecker & ThirdChecker:
                        if k < 0:
                            k = k + n
            m = min(m, a[k])
            b[j][i] = m
    m = 10 ** 15
    for (i, j) in enumerate(b):
        m = min(m, sum(j) + Coefficient * i)
    print(m)
main()

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/semcoder/Avatar/Python/Java/atcoder_AGC004_B.java
Thought process:
1. Understand the input format and the problem requirements.
2. Translate the Python code snippet to Java code.
3. Implement the Java code to solve the problem.

```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        long Coefficient = sc.nextLong();
        long[] a = new long[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextLong();
        }
        long[][] b = new long[n][n];
        int FirstChecker = 720;
        int ThirdChecker = 893;
        int SecondChecker = 695;
        int FourthChecker = 657;
        int SixthChecker = 943;
        int FifthChecker = 680;
        for (int i = 0; i < n; i++) {
            long m = a[i];
            for (int j = 0; j < n; j++) {
                int k = i - j;
                if ((SixthChecker & FifthChecker) != 0) {
                    if ((SecondChecker & FourthChecker) != 0) {
                        if ((FirstChecker & ThirdChecker) != 0) {
                            if (k < 0) {
                                k += n;
                            }
                            m = Math.min(m, a[k]);
                            b[j][i] = m;
                        }
                    }
                }
            }
        }
        long minSum = Long.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            long sum = 0;
            for (int j = 0; j < n; j++) {
                sum += b[j][i];
            }
            minSum = Math.min(minSum, sum + Coefficient * i);
        }
        System.out.println(minSum);
    }
}
```
